src/task/create-item-validation-task.ts

```diff
@@ -0,0 +1,142 @@
+
+    // get item
+    const item = await this.itemService.get(itemId, handler);
+
+    // recursively validate item and subitems
+    const validateItem = async (item: Item) => {
+
+      const executeProcess = async (process: ItemValidationProcess) => {
+        if (process?.name === ItemValidationProcesses.ImageChecking && item?.type !== this.serviceItemType)
+        return;
+
+        // create pending entry
+        const iVGEntry = await this.validationService.createItemValidationGroup(
+          item?.id,
+          iVId,
+          process.id,
+          getStatusIdByName(iVStatuses, ItemValidationStatuses.Pending),
+          handler,
+        ).then(data => data);
+
+        const status = await handleProcesses(process, item, this.fTM, this.actor, this.runner).catch(error => {
+          // log the error
+          console.error(error);
+          // if some error happend during the execution of a process, we consider the it failed
+          return ItemValidationStatuses.Failure;
+        });
+        
+        if (status === '')
+          throw new ProcessNotFoundError(process.name);
+
+        // need review if any process failed
+        if (status === ItemValidationStatuses.Failure) needReview = true;
+
+        console.log('prepare to update: ', iVGEntry.id);
+        // update status
+        const updatedEntry = await this.validationService.updateItemValidationGroup(iVGEntry.id, getStatusIdByName(iVStatuses, status), handler);
+        console.log('item validation group entry updated: ', updatedEntry);
+      };
+
+      // execute each process on item
+      await Promise.all(enabledProcesses.map(async (process) => {
+        await executeProcess(process).catch(error => {
+          throw(error(`Error occurs when executing process ${process.name}`));
+        });
+      }));
+
+      if (item?.type === ITEM_TYPE.FOLDER) {
+        const subItems = await this.itemService.getChildren(item, handler);
+        await Promise.all(subItems.map(async (subitem) => {
+          await validateItem(subitem).catch(error => {
+            throw(error(`Error occurs when validating item ${subitem?.name}`));
+          });
+        }));
+      }
+    };
+
+    await validateItem(item).catch(error => {
+      console.error(error);
+      // if error occurs, we would like a manual review on the item
+      needReview = true;
+    });
+
+    // create entry for review
+    const iVRStatuses = await this.validationService.getItemValidationReviewStatuses(handler);
+    if (needReview) await this.validationService.createItemValidationReview(
+      iVId,
+      getStatusIdByName(iVRStatuses, ItemValidationReviewStatuses.Pending),
+      handler,
+    );
+
+    this._message = 'Item validation task executed.';
+    // the result is only used for testing
+    this._result = needReview? 'pending manual review' : 'success';
+    // The task status is always 'OK', since the task itself completed successfully
+    this.status = 'OK';
+  }
+}
```

Comment:
Actually I'm really lost in your code, avoid defining very long functions inside other functions. When possible factor them out.